server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1558,17 +1558,28 @@ private CanMatchResponse canMatch(ShardSearchRequest request, boolean checkRefre
                     canMatch = aliasFilterCanMatch;
                 }
                 final FieldDoc searchAfterFieldDoc = getSearchAfterFieldDoc(request, context);
-                canMatch = canMatch && canMatchSearchAfter(searchAfterFieldDoc, minMax, sortBuilder);
+                canMatch = canMatch && canMatchSearchAfter(searchAfterFieldDoc, minMax, sortBuilder, request.source());
```

Comment:
You don't need to pass whole `source()`, right? 

```suggestion
                canMatch = canMatch && canMatchSearchAfter(searchAfterFieldDoc, minMax, sortBuilder, request.source().trackTotalHitsUpTo());
```

Reply:
Yeah, lets just pass totalHits....